src/Foundation/NSArray.cs

```diff
@@ -63,7 +63,44 @@ public static NSArray FromNSObjects (int count, params INativeObject [] items)
 		{
 			return FromNativeObjects (items);
 		}
+		public static NSArray FromNSObjects<T> (params T [][] items) where T: class, INativeObject
+		{
+			if (items == null)
+				return null;
+			
+			var ret = new NSMutableArray ((nuint)items.Length);
+			foreach (var t in items) {
+				if (t == null)
+					throw new ArgumentNullException(nameof (t));
+				foreach (var n in t) {
+					if(n == null)
+						throw new ArgumentNullException (nameof (n));
+				}
+
+				ret.Add (NSArray.FromNSObjects (t));	
+			}
 
+			return ret;
+		}
+		public static NSArray FromNSObjects<T> (T [,] items) where T: class, INativeObject
+		{
+			if (items == null)
+				return null;
+			
+			var width = items.GetLength (0);
+			var height = items.GetLength (1);
+			var ret = new T [height][];
+			for (var y = 0; y < height; y++) {
+				var row = new T [width];
+				for (var x = 0; x < width; x++) {
+					//if (items [x, y] == null)
+					//	throw new ArgumentNullException (); not sure if I should catch the error here 
```

Comment:
I need an advice here

Reply:
I don't think you need it as long as the other overload throws the correct exception.